data_juicer/ops/filter/face_area_filter.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+from jsonargparse.typing import ClosedUnitInterval
+
+from data_juicer.utils.availability_utils import AvailabilityChecking
+from data_juicer.utils.constant import Fields, StatsKeys
+from data_juicer.utils.mm_utils import load_image, pil_to_opencv
+
+from ..base_op import OPERATORS, Filter
+from ..op_fusion import LOADED_IMAGES
+
+OP_NAME = 'face_area_filter'
+
+with AvailabilityChecking(['dlib'], OP_NAME):
+    import dlib
+
+
+@OPERATORS.register_module(OP_NAME)
+@LOADED_IMAGES.register_module(OP_NAME)
+class FaceAreaFilter(Filter):
+    """Filter to keep samples with face area ratio within a specific range.
+    """
+
+    def __init__(self,
+                 min_ratio: ClosedUnitInterval = 0.0,
+                 max_ratio: ClosedUnitInterval = 0.4,
+                 any_or_all: str = 'any',
+                 *args,
+                 **kwargs):
+        """
+        Initialization method.
+
+        :param min_ratio: Min ratio for the largest face area in an image.
+        :param max_ratio: Max ratio for the largest face area in an image.
+        :param any_or_all: Keep this sample with 'any' or 'all' strategy of
+            all images. 'any': keep this sample if any images meet the
+            condition. 'all': keep this sample only if all images meet the
+            condition.
+        :param args: Extra positional arguments.
+        :param kwargs: Extra keyword arguments.
+        """
+
+        # Extract face detector arguments from kwargs
+        detector_keys = ['upsample_num_times']
+        self.detector_kwargs = {
+            key: kwargs.pop(key)
+            for key in detector_keys if key in kwargs
+        }
+
+        super().__init__(*args, **kwargs)
+        self.min_ratio = min_ratio
+        self.max_ratio = max_ratio
+
+        if any_or_all not in ['any', 'all']:
+            raise ValueError(f'Keep strategy [{any_or_all}] is not supported. '
+                             f'Can only be one of ["any", "all"].')
+        self.any = (any_or_all == 'any')
+
+        # Initialize face detector
+        self.detector = dlib.get_frontal_face_detector()
+
+    def compute_stats(self, sample, context=False):
+        # check if it's computed already
+        if StatsKeys.face_ratios in sample[Fields.stats]:
+            return sample
+
+        # there is no image in this sample, still default ratio 0.0
+        if self.image_key not in sample or not sample[self.image_key]:
+            sample[Fields.stats][StatsKeys.face_ratios] = [0.0]
```

Comment:
Setting it to 0 for samples without images might influence the overall distribution of face ratios. How about setting it to an empty float64 array like other OPs?
```
sample[Fields.stats][StatsKeys.image_text_matching_score] = np.array([], dtype=np.float64)
```